plugins/pulp_rpm/plugins/importers/yum/pulp_solv.py

```diff
@@ -321,299 +264,285 @@ def rpm_dependency_conversion(solvable, unit_name, unit_flags, unit_evr,
+        but in this method, we are only constructing the condition after the "Provides:"
+        or "Requires:". This method creates such solve.Dep().
+        """
+        return pool.Dep("module({}:{})".format(name, stream))
+
+    def dep_or_rel(dep, op, rel):
+        """
+        There are relations between modules in `deps`. For example:
+          deps = [{'gtk': ['1'], 'foo': ['1']}]" means "gtk:1 and foo:1" are both required.
+          deps = [{'gtk': ['1', '2']}"] means "gtk:1 or gtk:2" are required.
+
+        This method helps creating such relations using following syntax:
+          dep_or_rel(solv.Dep, solve.REL_OR, stream_dep(name, stream))
+          dep_or_rel(solv.Dep, solve.REL_AND, stream_dep(name, stream))
+          dep_or_rel(solv.Dep, solve.REL_WITH, stream_dep(name, stream))
+          dep_or_rel(solv.Dep, solve.REL_WITHOUT, stream_dep(name, stream))
+        """
+        dep.Rel(op, rel) if dep is not None else rel
+
+    # Check each dependency dict in dependency_list and generate the solv requirements.
+    reqs = None
+    for dep_dict in dependency_list:
+        require = None
+        for name, streams in dep_dict.items():
+            if name == 'platform':
+                # no need to fake the platform (streams) later on
+                continue
+            name = name.encode('utf8')
+
+            # The req_pos will store solv.Dep expression for "positive" requirements.
+            # That is the case of 'gtk': ['1', '2'].
+            # The req_neg will store negative requirements like 'gtk': ['-1', '-2'].
+            req_pos = req_neg = None
+
+            # For each stream in `streams` for this dependency, generate the
+            # module(name:stream) solv.Dep and add REL_OR relations between them.
+            for stream in streams:
+                stream = stream.encode('utf8')
+
+                if stream.startswith("-"):
+                    req_neg = dep_or_rel(req_neg, solv.REL_OR, stream_dep(name, stream[1:]))
+                else:
+                    req_pos = dep_or_rel(req_pos, solv.REL_OR, stream_dep(name, stream))
+
+            # Generate the module(name) solv.Dep.
+            req = pool.Dep("module({})".format(name))
+
+            # Use the REL_WITH for positive requirements and REL_WITHOUT for negative
+            # requirements.
+            if req_pos is not None:
+                req = req.Rel(solv.REL_WITH, req_pos)
+            elif req_neg is not None:
+                req = req.Rel(solv.REL_WITHOUT, req_neg)
+
+            # And in the end use AND between the last name:[streams] and the current one.
+            require = dep_or_rel(require, solv.REL_AND, req)
+
+        # NOTE: In the original algorithm, this was an OR operation. We don't want only one
+        # set of deps (for one platform), we want the deps for all platforms. Hence, solv.REL_AND.
+        reqs = dep_or_rel(reqs, solv.REL_AND, require)
```

Comment:
I'm not sure I fully understand this.
Since we don't take into account platforms and would like to have all the dependencies, can it be the problem anywhere down the line to have conflicting sets having relation `solv.REL_AND`?

Reply:
@goosemania Could you give an example?

Reply:
No :)
I can try and tell me if it's a valid example or not.
Set of deps for platformA: dep1, dep2.
Set of deps for platformB: dep2, dep3.
dep1 and dep3 are in conflict with each other.

As far as I understood, you are combining those two set with AND and pass down the line that all of them are required. So my question was, would it cause any problem later in libsolv depsolving process if dep1 and dep3 are conflicting.

Reply:
I don't think so.  This part of the code should be equivalent to what was here before.

(before, it was calling ```solvable.add_deparray()``` inside the loop for each dep dict/platform)

Reply:
It's weird because Pulp pretends to be both a client and not-a-client.  We can have multiple versions of each package installed simultaneously for instance.

Reply:
Ok, if that works, then ok.